runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -67,11 +75,22 @@ MM_IndexableObjectAllocationModel::initializeAllocateDescription(MM_EnvironmentB
 		break;
 
 	case GC_ArrayletObjectModel::InlineContiguous:
-		/* all good */
-		setAllocatable(true);
+		/* Check if we're dealing with a camouflaged discontiguous array - these arrays will require slow-path allocate */
+		if (isAllIndexableDataContiguousEnabled && (!extensions->indexableObjectModel.isArrayletDataAdjacentToHeader(_dataSize))) {
+			if (isGCAllowed()) {
```

Comment:
needs more comments why/why layoutSizeInBytes is set

Reply:
`layoutSizeInBytes` is used to track amount of array bytes not included to spine and set final total number of bytes allocated for array:
```
_allocateDescription.setBytesRequested(spineBytes + layoutSizeInBytes); <--- in spine plus out of spine
```
This logic mimics out-of-spine behaviour for Discontiguous/Hybrid. `allocDescription->getBytesRequested()` is used across the code as total amount of memory required for this allocation. For example, it is used to report amount failed to be allocated in case of OOM.

Reply:
thanks @dmitripivkine
@LinHu2016 please add a comment based on this comment